contracts/adapters/AaveLeverageStrategyExtension.sol

```diff
@@ -0,0 +1,1275 @@
+     * the leverage token will move towards the target leverage each rebalance.
+     *
+     * return uint256          New leverage ratio
+     */
+    function _calculateNewLeverageRatio(uint256 _currentLeverageRatio) internal view returns(uint256) {
+        // CLRt+1 = max(MINLR, min(MAXLR, CLRt * (1 - RS) + TLR * RS))
+        // a: TLR * RS
+        // b: (1- RS) * CLRt
+        // c: (1- RS) * CLRt + TLR * RS
+        // d: min(MAXLR, CLRt * (1 - RS) + TLR * RS)
+        uint256 a = methodology.targetLeverageRatio.preciseMul(methodology.recenteringSpeed);
+        uint256 b = PreciseUnitMath.preciseUnit().sub(methodology.recenteringSpeed).preciseMul(_currentLeverageRatio);
+        uint256 c = a.add(b);
+        uint256 d = Math.min(c, methodology.maxLeverageRatio);
+        return Math.max(methodology.minLeverageRatio, d);
+    }
+
+    /**
+     * Calculate total notional rebalance quantity and chunked rebalance quantity in precise (18 decimal) units. 
+     *
+     * return uint256          Chunked rebalance notional in collateral units
+     * return uint256          Total rebalance notional in collateral units
+     */
+    function _calculateChunkRebalanceNotional(
+        LeverageInfo memory _leverageInfo,
+        uint256 _newLeverageRatio,
+        bool _isLever
+    )
+        internal
+        view
+        returns (uint256, uint256)
+    {
+        // Calculate absolute value of difference between new and current leverage ratio
+        uint256 leverageRatioDifference = _isLever ? _newLeverageRatio.sub(_leverageInfo.currentLeverageRatio) : _leverageInfo.currentLeverageRatio.sub(_newLeverageRatio);
+
+        uint256 totalRebalanceNotional = leverageRatioDifference.preciseDiv(_leverageInfo.currentLeverageRatio).preciseMul(_leverageInfo.action.collateralBalance);
+
+        uint256 maxBorrow = _calculateMaxBorrowCollateral(_leverageInfo.action, _isLever);
+
+        uint256 chunkRebalanceNotional = Math.min(Math.min(maxBorrow, totalRebalanceNotional), _leverageInfo.twapMaxTradeSize.mul(10 ** (18 - strategy.collateralDecimals)));
+
+        return (chunkRebalanceNotional, totalRebalanceNotional);
+    }
+
+    /**
+     * Calculate the max borrow / repay amount allowed in precise units for lever / delever. This is due to overcollateralization requirements on
+     * assets deposited in lending protocols for borrowing.
+     * 
+     * For lever, max borrow is calculated as:
+     * (Net borrow limit in USD - existing borrow value in USD) / collateral asset price adjusted for decimals
+     *
+     * For delever, max borrow is calculated as:
+     * Collateral balance in precise units * (net borrow limit in USD - existing borrow value in USD) / net borrow limit in USD
+     *
+     * Net borrow limit for levering is calculated as:
+     * The collateral value in USD * Aave collateral factor * (1 - unutilized leverage %)
+     *
+     * Net borrow limit for delevering is calculated as:
```

Comment:
There is an inconsistency between javadocs here. At yout beginning you have mentioned borrow/repay for lever/delever. But here you have written borrow for delever as well.